src/main/scala/com/kinja/amqp/persistence/MessageStore.scala

```diff
@@ -48,48 +43,19 @@ trait MessageStore {
 	def deleteFailedMessage(id: Long): Future[Unit]
 
 	/**
-	 * Delete single confirmations matching some messages, along with those messages
-	 *
-	 * @return Number of removed confirmations, which is also the number of removed messages
-	 */
-	def deleteMatchingMessagesAndSingleConfirms(): Future[Int]
-
-	/**
-	 * Delete messages that were confirmed by some multiple confirmation
-	 *
-	 * @return Number of messages deleted
-	 */
-	def deleteMessagesWithMatchingMultiConfirms(): Future[Int]
-
-	/**
-	 * Delete old multiple confirmations that do not match any messages
-	 *
-	 * @return Number of confirmations deleted
-	 */
-	def deleteMultiConfIfNoMatchingMsg(): Future[Int]
-
-	/**
-	 * Delete old single confirmations
-	 *
-	 * @return Number of confirmations deleted
-	 */
-	def deleteOldSingleConfirms(): Future[Int]
-
-	/**
-	 * Lock some messages to this host
+	 * Delete confirmed messages, as well as matching single confirmations
+	 * and confirmations that are too old
 	 *
-	 * @param limit How many messages to lock
-	 * @return Number of messages locked
+	 * @return Number of removed messages
 	 */
-	def lockOldRows(limit: Int): Future[Int]
+	def cleanup(): Future[Boolean]
 
 	/**
-	 * Load messages that were locked to this host
+	 * Lock and load an appropriate number of old messages
 	 *
-	 * @param limit How many messages to load
 	 * @return List of locked messages
 	 */
-	def loadLockedMessages(limit: Int): Future[List[MessageLike]]
+	def lockAndLoad(): Future[List[MessageLike]]
```

Comment:
Possible implementation, as if using the old interface:

	override def lockAndLoad(): Future[List[MessageLike]] = for {
		_ <- messageStore.lockOldRows(batchSize)
		// in case we have more locked rows than the batch size
		// (failed to process after previous lock)
		messages <- messageStore.loadLockedMessages(batchSize * 2)
	} yield messages